cloud/amazon/ec2_vpc.py

```diff
@@ -406,9 +407,37 @@ def create_vpc(module, vpc_conn):
         # First add all new subnets
         for subnet in subnets:
             add_subnet = True
+            subnet_tags_current = True
+            new_subnet_tags = subnet.get('resource_tags', None)
+            subnet_tags_delete = []
+            
             for csn in current_subnets:
                 if subnet['cidr'] == csn.cidr_block:
                     add_subnet = False
+
+                # Remove any subnet tags not in playbook
+                subnet_tags_extra = (set(csn.tags.items()).issubset(set(new_subnet_tags.items())))
+                if subnet_tags_extra == False:
+                    try:
+                        for item in csn.tags.items():
+                            if item not in new_subnet_tags.items():
+                                subnet_tags_delete.append(item)
+
+                        subnet_tags_delete = [key[0] for key in subnet_tags_delete]
+                        delete_subnet_tag = vpc_conn.delete_tags(csn.id, subnet_tags_delete)
+                        changed = True
+                    except EC2ResponseError, e:
+                        module.fail_json(msg='Unable to delete resource tag, error {0}'.format(e))
+                # Add new subnet tags if not current
+                subnet_tags_current = (set(new_subnet_tags.items()).issubset(set(csn.tags.items())))
+                if subnet_tags_current != True:
```

Comment:
Awesome. So just to clarify, the intended context is that the subnet tags are deleted only if there are no tags to be added?